### PR TITLE
Prevent invalid date in DateView

### PR DESCRIPTION
**Pull request by Henry0422**
Fixed the issue that DateView accept `2019/02/90` issue using **Safari**.

### CODE REVIEW

File: src/foam/u2/DateView.js [hunk 1]
```diff
@@ -34,19 +38,36 @@ foam.CLASS({
...
+  listeners: [
+    function checkDateFormat(value) {
+      oldDate = value.split(/[^0-9]/).map((str) => {
+        return parseInt(str, 10);
+      }).join('');
+
+      var newDate = new Date(oldDate);
+      newDate = `${newDate.getFullYear()}${newDate.getMonth() + 1}${newDate.getDate()}`;
+      if ( newDate !== oldDate ) {
+        ctrl.notify('Invalid Date.', 'error');
```
Comment (zahnno) [inline src/foam/u2/DateView.js:68]: Please remove the notify and replace with a console error.

Comment (Henry0422) [reply src/foam/u2/DateView.js:68]: Console.error won't be a good way. We need a way to notify the user their input is not appropriate

File: src/foam/u2/DateView.js [hunk 2]
```diff
@@ -34,19 +38,36 @@ foam.CLASS({
...
           function(value) {
             return new Date(value);
           }
       );
     }
+  ],
+
+  listeners: [
+    function checkDateFormat(value) {
+      oldDate = value.split(/[^0-9]/).map((str) => {
```
Comment (TW80000) [inline src/foam/u2/DateView.js:61]: Missing `var`

**Comment by Henry0422** [comment]
OK, this might break the other DateView.....

**Review by TW80000 — CHANGES_REQUESTED** [review]

**Comment by TW80000** [comment]
I don't think I like the approach in this PR anymore. The best solution to this problem would be to create a FOAM date-picker that would be used in all browsers and wouldn't let users select invalid dates in the first place.

**Comment by jlhughes** [comment]
I’m in agreement with @TW80000. It does seem absurd that a user can pick an invalid date.  With regard to launch, since it’s beta, we can restrict browser choice, and then deal with this issue properly which is mostly likely a FOAM date picket.

**Comment by kgrgreer** [comment]
FOAM1 has /js/foam/u2/md/DatePicker.js

**Pull request closed without merging**